### PR TITLE
Split the Extensions page in WP Admin and add respective menu items

### PR DESCRIPTION
**Pull request by madeincosmos**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
- Register a new top-level menu for Marketplace and create two subpages for Extensions and My Subscriptions
- register new page IDs for loading WooCommerce scripts and styles
- Add a method in `WC_Admin_Addons` for Helper page output only
- remove top nav from the Extensions page

### How to test the changes in this Pull Request:

1. Check out this branch on local environment.
2. Go to WP Admin and observe there's a new top-level menu called Marketplace with two subpages called Extensions and My Subscriptions.
3. Open each of these pages and make sure they load correctly and have all styles in place.
4. Make sure the correct menu item is highlighted when you open each page.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Notes
- didn't remove top nav on the My Subscriptions page yet because editing this file triggered a lot of phpcs errors. I'll take care of it later.

### CODE REVIEW

File: includes/admin/helper/views/html-oauth-start.php [hunk 1]
```diff
@@ -10,7 +10,6 @@
 ?>
 
 <div class="wrap woocommerce wc_addons_wrap wc-helper">
-	<?php require WC_Helper::get_view_filename( 'html-section-nav.php' ); ?>
```
Comment (kloon) [inline includes/admin/helper/views/html-oauth-start.php:13]: Should we remove this file if we are not going to be using the tabs anymore?

File: includes/admin/helper/class-wc-helper.php [hunk 2]
```diff
@@ -466,9 +466,10 @@ private static function _filter( &$subscriptions, $filter ) {
 	public static function admin_enqueue_scripts() {
 		$screen       = get_current_screen();
 		$screen_id    = $screen ? $screen->id : '';
-		$wc_screen_id = sanitize_title( __( 'WooCommerce', 'woocommerce' ) );
 
-		if ( $wc_screen_id . '_page_wc-addons' === $screen_id && isset( $_GET['section'] ) && 'helper' === $_GET['section'] ) {
+		if ( 'marketplace_page_wccom-subscriptions' == $screen_id || (
```
Comment (kloon) [inline includes/admin/helper/class-wc-helper.php:470]: Can we improve the readability of this statement a bit? Perhaps breaking it up into more one liners? You were heading in the right direction.

File: includes/admin/class-wc-admin-menus.php [hunk 3]
```diff
@@ -147,10 +148,14 @@ public function status_menu() {
 	 * Addons menu item.
 	 */
 	public function addons_menu() {
+		add_menu_page( __( 'Marketplace', 'woocommerce' ), __( 'Marketplace', 'woocommerce' ), 'manage_woocommerce', 'wc-addons', null, null, '58' );
```
Comment (kloon) [inline includes/admin/class-wc-admin-menus.php:151]: Do we have an icon for this menu?

**Review by kloon — APPROVED** [review]
Changes look good Maria, just left a small couple of comments.

The biggest blocker would be to update the E2E tests as well for this, we can decide to do this at a later stage or in this PR if you want.

The test in question is https://github.com/woocommerce/woocommerce/blob/trunk/tests/e2e/core-tests/specs/merchant/wp-admin-extensions-connect-wccom.test.js#L18

**Comment by kloon** [comment]
OK, so just an update, based on a conversation in slack the menu structure is wrong here.

`WooCommerce -> Extensions` needs to be replaced with `WooCommerce -> Marketplace` and then we need to move the subscriptions to `WooCommerce -> My Subscriptions`

File: includes/admin/helper/views/html-oauth-start.php (hunk 1 above)
Comment (madeincosmos) [reply includes/admin/helper/views/html-oauth-start.php:13]: Removed it now.

File: includes/admin/helper/class-wc-helper.php (hunk 2 above)
Comment (madeincosmos) [reply includes/admin/helper/class-wc-helper.php]: Reverted this now as we're back to `woocommerce` screen id.

File: includes/admin/class-wc-admin-menus.php (hunk 3 above)
Comment (madeincosmos) [reply includes/admin/class-wc-admin-menus.php]: We removed the menu for now, so the icon won't be necessary.

**Comment by andfinally** [comment]
Looks like we still need to rename the Extensions menu item to Marketplace.

**Comment by kloon** [comment]
Apart from the issue @andfinally mentions, also noticed that subscription notice bubble is showing on the wrong menu item now, that bubble should show on My Subscriptions

**Comment by madeincosmos** [comment]
Closing in favor of #30380

**Pull request closed without merging**